### ISSUE TITLE
1 hit kill?

### ISSUE THREAD

**Issue by EnderKenji** [Minifixio/pocketmine-1vs1-plugin#21]
Sometimes Peeps joined , and start duel, 

A person hit B person then B person won , even he doesnt hit anyone. And its 1 hit kill...

Im using ImagicalMine, is that related?

**Comment by robske110** [comment]
that's not related to this plugin (IM)

**Comment by Minifixio** [comment]
Wow I don't know what is that. I think it's something from ImaginalCorp because I've changed anything about the hits or something like that... I'll search but I promise nothing.

**Comment by EnderKenji** [comment]
im pretty sure its 1v1 prob, cus only start a duel will happen this... o.0

**Comment by robske110** [comment]
I am using an older version of ImagicalMine and there it works!
I think its because of setHealth does not work correctly!

**Comment by ZombiePower** [comment]
I don't think, its a problem with ImagicalMine, as we haven't changed it that much! xD

**Comment by robske110** [comment]
You have completely messed up the Damage/Death system!
And its an known bug that setHealth doesn't seem to work correctly!

**Comment by ghost** [comment]
Yeah. That's what i want to say. When i join 1vs1 and my lfe is not full.
- its decrease when the battle start
- when i won im gonna die and i cant click respawn
- when i get hit and i won my life isnt full.

**Comment by robske110** [comment]
IMAGICALMINE
IT IS NOT THE PLUGIN FAULT
IF YOU WOULD UNDERSTAND ANYTHING OF PHP THEN LOOK IN THE SOURCE CODE AND SAY ME WHERE AN ERROR IS?

**Comment by Minifixio** [comment]
Okay I think robske is right. PocketMine do not have all the ImaginalMine possibilities for now.

**Comment by ghost** [comment]
So its pocketmine fault ?. where can i get imagicalmine?

**Comment by robske110** [comment]
Yes its pocketmine/imagicalmine fault
With the setHealth function

**Comment by ghost** [comment]
Aaa. I dont get it. Its pocketmine and imagicalmine Fault ??

**Comment by robske110** [comment]
Yes both have problems with the respawn screen and with the health management IM has messed it up and then fixed it so it's fixed there but messed up. In pocketmine IDK...

**Comment by Minifixio** [comment]
PocketMine is very incomplete and ImaginaMine is building by the community so both have their bugs. So both are bugged but ImaginalMine is better. It's why sometimes there is some bugs with my plugins because I started to develop it on PocketMine and now I have to pass on ImaginalMine.

**Comment by ZombiePower** [comment]
I think, you should develop plugin based of ImagicalMine :)

**Comment by robske110** [comment]
What? ImagicalMine has changed nothing in the API, only added Hunger and XP
@minifixio Please check out my pull request
